packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -249,6 +241,61 @@ export interface GraphQLApiProps {
 
 }
 
+/**
+ * A IamResource Class for IAM Permissions
+ */
+export class IamResource {
+  /**
+   * Generate the resourceNames given custom arns
+   *
+   * @param arns The custom arns that need to be permissioned
```

Comment:
avoid the casing as it won't be idiomatic for every language. Describe the purpose of the function to help a user decide if it achieves what they want rather than the outcome of calling the API. maybe even include an example of a custom arn for the param annotation `(i.e. <example>`